### PR TITLE
Add mapbox dependencies

### PR DESCRIPTION
**Pull request by anselmbradford**
Moves mapbox into new template with minimal cleanup, so future cleanup is easier to review.

## Additions

- Adds mapbox map and dependencies to improved finding-a-housing-counselor template.

## Testing

1. `./frontend.sh` and http://localhost:8000/find-a-housing-counselor/?zipcode=05201 should show a map

## Screenshots

screen shot 2018-11-15 at 1 30 40 pm

### CODE REVIEW

**Comment by anselmbradford** [comment]
Note there were some unrelated files that the linter fixed that I decided to just leave in because it's minor whitespace fixes.

**Comment by mistergone** [comment]
The form submission doesn't work because the `name` in the `INPUT` is `"zip"` - changing it to `name="zipcode"` fixes this. Not sure why/where this changed in the backend, but uh... changing it makes it work!

File: cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html [hunk 1]
```diff
@@ -180,3 +190,25 @@ <h3>Paperwork Reduction Act statement</h3>
         </div>
     </main>
 {% endblock %}
+
+{% block javascript %}
+    {{ super() }}
+    <script type='text/javascript'
+            src='https://api.mapbox.com/mapbox.js/v2.4.0/mapbox.js'></script>
+    <script>
+    var hud_data = {{ api_json | tojson }};
```
[the remark this answers is not in the document]
Comment (anselmbradford) [reply cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html]: Fixed!

**Comment by mistergone** [comment]
I'm having problems locally with mapbox, where the points are placed on the "map", but no image for the map appear:

screen shot 2018-11-15 at 4 56 01 pm

The console error is:
```
util.js:19 could not load TileJSON at http://a.tiles.mapbox.com/v4/mapbox.streets.json?access_token=None
```

**Pull request merged**